core/messagesource.go

```diff
@@ -22,6 +22,9 @@ type MessageSource interface {
 
 	// IsBlocked returns true if the source cannot produce messages
 	IsBlocked() bool
+
+	// GetID returns the pluginID of the message source
+	GetID() string
```

Comment:
As GetID is shared between plugin and message source we should make this an additional interface